docs/learn/introduction/04-the-domain-level.md

```diff
@@ -0,0 +1,351 @@
+    // We leverage the forEachActiveCell method to easily iterate over the active cells.
+    sphereSdf.forEachActiveCell([&](const Neon::index_3d& idx, int, double& value) {
+        double sdf = sdfCenteredSphere(idx, dim, voxelEdge, r);
+        value = sdf;
+    });
+
+    // Exporting some information of the sdf on terminal and on a vtk file. 
+    NEON_INFO(sphereSdf.toString());
+    sphereSdf.ioToVtk("sdf", "sdf");
+    
+    return 0;
+}
+```
+
+![](img/04-initial-sdf.png)
+
+<a name="mapContainers">
+### **Neon map containers**: expanding the sphere via a level set
+</a>
+
+Let's now manipulate the sdf to expand zero level set of the sphere.
+The operation can be simply implemented by adding a constant to all the cells of the grid.
+The result will be a level set field (not a signed distance function).
+
+Until now the values of the `sphereSdf` field are stored on the host,
+indeed working on the host side is the only way we can do IO type of operations.
+The process of moving data to the devices is simple as it requires only a simple mehtod call.
+
+!!! Warning
+
+    All Neon operation related to runnig computation or moving data have a asynchronous semantic.
+    At the domain level it is up to the user to handle any sort of synchronization manually. 
+
+```cpp linenums="93" title="Neon/tutorials/introduction/domainLevel/domainLevel.cpp"
+    // ...
+    
+    // Step 4 -> Neon map containers: expanding the sphere via a level set
+
+    // loading the sphereSdf to device
+    sphereSdf.updateCompute(Neon::Backend::mainStreamIdx);
+
+    // Run a container that ads a value to the sphere sdf
+    // The result is a level set of an expanded sphere (not more a sdf)
+    // We run the container asynchronously on the main stream
+    expandedLevelSet(sphereSdf, 5.0).run(Neon::Backend::mainStreamIdx);
+
+    // Moving asynchronously the values of the newly computed level set back
+    // to export the result to vtk.
+    sphereSdf.updateIO(Neon::Backend::mainStreamIdx);
+
+    // Waiting for the transfer to complete.
+    backend.sync(Neon::Backend::mainStreamIdx);
+
+    // Exporting once again the fiel to vtk
+    sphereSdf.ioToVtk("expandedLevelSet", "expandedLevelSet");
+    
+    return 0;
+}
+```
+
+```cpp linenums="6" title="Neon/tutorials/introduction/domainLevel/expandSphere.cu"
+template <typename Field>
+auto expandedLevelSet(Field& sdf,
+                 double expantion)
+    ->Neon::set::Container
+{
+    return sdf.getGrid().getContainer(
+        "ExpandedLevelSet", [&, expantion](Neon::set::Loader& L) {
+            auto& px = L.load(sdf);
+
+            return [=] NEON_CUDA_HOST_DEVICE(
+                       const typename Field::Cell& cell) mutable {
+                px(cell, 0) -= expantion;
+            };
+        });
+}
+```
+
+![](img/04-expanded-levelSet.png)
+
+<a name="mapContainstencilContainersers">
+### Neon stencil containers: computing the grad of the level set field
```

Comment:
Can we say couple of words about the stencil operations implemented here?

Reply:
Agree, and that is the plan. This section of the tutorial has not been completed yet.